app/workers/detect_invariants.rb

```diff
@@ -97,7 +97,7 @@ def detect_applications_with_course_choices_in_previous_cycle
   def detect_submitted_applications_with_more_than_three_course_choices
     applications_with_too_many_choices = ApplicationForm
       .joins(:application_choices)
-      .where.not(application_choices: { status: 'unsubmitted' })
+      .where(application_choices: { status: ApplicationStateChange::DECISION_PENDING_STATUSES || ApplicationStateChange::ACCEPTED_STATES })
```

Comment:
So i think this actually needs to be

`
.where(application_choices: { status: (ApplicationStateChange::DECISION_PENDING_STATUSES + ApplicationStateChange::ACCEPTED_STATES) })
`